src/validation/mo_sethet/sethet.cpp

```diff
@@ -101,34 +102,32 @@ void sethet(Ensemble *ensemble) {
     so2_diss = haero::testing::create_column_view(pver);
 
     ColumnView tmp_hetrates[gas_pcnst];
-    ColumnView qin[gas_pcnst];
     View1DHost tmp_hetrates_host[gas_pcnst];
-    View1DHost qin_host[gas_pcnst];
+    View2DHost qin_host("qin_host", pver, gas_pcnst);
 
     View2D het_rates("het_rates", pver, gas_pcnst);
+    View2D qin("qin", pver, gas_pcnst);
     auto het_rates_host = Kokkos::create_mirror_view(het_rates);
 
     for (int mm = 0; mm < gas_pcnst; ++mm) {
 
       tmp_hetrates[mm] = haero::testing::create_column_view(pver);
-      qin[mm] = haero::testing::create_column_view(pver);
       tmp_hetrates_host[mm] = View1DHost("tmp_hetrates_host", pver);
-      qin_host[mm] = View1DHost("qin_host", pver);
     }
 
     int count = 0;
     for (int mm = 0; mm < gas_pcnst; ++mm) {
       for (int kk = 0; kk < pver; ++kk) {
-        qin_host[mm](kk) = qin_in[count];
+        qin_host(kk, mm) = qin_in[count];
```

Comment:
Whenever we are switching between arrays and views, we should test it on Frontier so that the model doesn't break on it.